### ISSUE TITLE
Series upgrade: SZ did not extract the embedded SRT subtitles (2)

### ISSUE THREAD

**Issue by cphmichael** [pannal/Sub-Zero.bundle#692]
I have attached data from the latest 4 series that are updated:

* Emergence S01E05 - Updated 1 November 2019 15:17
* Treadstone S01E03 - Updated 1 November 2019 11:55
* Prodigal Son S01E06 - Updated 31 October 2019 12:39
* Batwoman S01E04 - Updated 31 October 2019 05:54

In Kitana 'History' they are all shown with there first upload date. Not the update date.

If you look at Emergence S01E05 - it says:
* auto, English in TV Shows (embedded, score: 0), 2019-10-30 19:03

If I click on the Emergence S01E05 - Kitana says 'No current subtitle in storage'.

It is the same for pattern for the other updated series.

[SZ.zip](https://github.com/pannal/Sub-Zero.bundle/files/3799265/SZ.zip)

![Sonarr - Emergence S01E05](https://user-images.githubusercontent.com/45288008/68049584-963f0900-fce3-11e9-97e7-5107802171aa.png)

![Sonarr - Treadstone S01E03](https://user-images.githubusercontent.com/45288008/68049605-a0610780-fce3-11e9-8f9f-104ff6fe386a.png)

![Sonarr - Prodigal Son S01E06](https://user-images.githubusercontent.com/45288008/68049623-a9ea6f80-fce3-11e9-8134-8ef4896f73b5.png)

![Sonarr - Batwoman S01E04](https://user-images.githubusercontent.com/45288008/68049638-b79ff500-fce3-11e9-8e92-9f235bf7b55b.png)

![PMS - Emergence S01E05](https://user-images.githubusercontent.com/45288008/68049463-406a6100-fce3-11e9-8378-232806a3a9de.png)

![PMS - Treadstone S01E03](https://user-images.githubusercontent.com/45288008/68049479-495b3280-fce3-11e9-837f-22b750ca5f9b.png)

![PMS - Prodigal Son S01E06](https://user-images.githubusercontent.com/45288008/68049488-50824080-fce3-11e9-9bb8-b3a5dfa666b9.png)

![PMS - Batwoman S01E04](https://user-images.githubusercontent.com/45288008/68049496-58da7b80-fce3-11e9-807a-ab1170edb7de.png)

![Kitana - Emergence S01E05 (1)](https://user-images.githubusercontent.com/45288008/68049720-e322df80-fce3-11e9-8f3d-d514f6b28e3e.png)

![Kitana - Emergence S01E05 (2)](https://user-images.githubusercontent.com/45288008/68049736-eae28400-fce3-11e9-95fa-ce346a96fcf8.png)

![Kitana - Treadstone S01E03 (1)](https://user-images.githubusercontent.com/45288008/68049770-fa61cd00-fce3-11e9-9d43-f32c1b19bde9.png)

![Kitana - Prodigal Son S01E06 (1)](https://user-images.githubusercontent.com/45288008/68049950-6e03da00-fce4-11e9-92d6-76ec7affb5a6.png)

![Kitana - Batwoman S01E04 (1)](https://user-images.githubusercontent.com/45288008/68049793-0b124300-fce4-11e9-99db-3c64e0c5b3cf.png)

**Comment by pannal** [comment]
> In Kitana 'History' they are all shown with there first upload date. Not the update date.
> If you look at Emergence S01E05 - it says:

> auto, English in TV Shows (embedded, score: 0), 2019-10-30 19:03
> If I click on the Emergence S01E05 - Kitana says 'No current subtitle in storage'.

Yes, those are historic entries. The part ID changes once the physical media is upgraded, so on the next cleanup run SZ detects that the stored info for the part ID that was used before is non existant and cleans the storage. This happens regularly. 

That doesn't kill the historic entries, though.

If the issue persists and upgrades aren't properly detected, I might add several other "embedded extraction points" in the cleanup and scheduler phase.

**Comment by pannal** [comment]
Subtitle storage maintenance runs every 7 days, FindBetter runs periodic, as defined by the user, SearchAllRecentlyAddedMissing as well.

I think this might be the only choice for us here, as the bug reports that I've posted in the internal Plex bug tracker have been widely ignored.

**Comment by pannal** [comment]
If you have the same settings set as in #691, essentially ignoring the whole SZ process once you have one matching embedded subtitle in a media file, this will not do anything.

**Comment by cphmichael** [comment]
I have not changed anything i the settings.

I believe that the embedded subtitles are a better match than an external downloaded subtitles.

Therefore I prefer to extract the embedded subtitles with configured default mods over downloading an external subtitles.

If SZ is not able to detect when an item is upgraded and therefore not extracting the embedded subtitles, my thought was that the scheduler could do the job :-)

**Comment by pannal** [comment]
> my thought was that the scheduler could do the job :-)

Yes, it does now.

SZ can only react to what the PMS is doing. If the agent isn't called upon an upgrade, SZ doesn't know that anything's changed.
